abi/decode.go

```diff
@@ -15,11 +15,15 @@ const (
 	storageSlotSize = 32
 )
 
-// Decode decodes the input with a given type
-func Decode(t *Type, input []byte) (interface{}, error) {
+// Decode decodes the input with a given type“
+func Decode(t *Type, input []byte, isFunctionArgument ...any) (interface{}, error) {
```

Comment:
Sorry for my ignorance, but why this couldn't be of the `bool` type?

```suggestion
func Decode(t *Type, input []byte, isFunctionArg bool) (interface{}, error) {
```